### ISSUE TITLE
Get rid of required "Site address" settings

### ISSUE THREAD

**Issue by rubendel** [opensourceBIM/BIMserver#698]
This setting is a cause of much confusion. It would make deployment a lot easier if this settings was not required.

Webclients should find another way to determine the server's API address (for example detect the "/apps/" part of the URL).

To test:
- Deployment on non-root-context path (e.d. http://domain/bimserverwar-1.5.95)
- HTTPS

**Comment by rubendel** [comment]
This setting is also used for the base address for links that are put into emails, so for this reason alone the setting should stay.

I think most of the problems would be fixed by making BIMvie.ws embed both BIMsurfer and the BIMserverJavaScript API during the build. Also this would make the webmodules cleaner (no more issues about BIMsurfer not working from people clicking on the BIMsurfer webmodule...)

Also used in:
- When creating a BCF (ClashDetection, IfcValidator), reference links are built up using the SiteAddress (they are however invalid, but in the future this would be required non-the-less)
- When sending notifications, the SiteAddress is used as the base address for (returning) API calls
- OAuth uses it to determine resourceUrl etc...
- A few more places, for example the internal BIMserver client uses it to determine checkin/download URL etc...

So in short, the setting itself should not go, but for all those people getting the "BIMvie.ws is loading" message (forever), we could make it better by creating an embedded build so that it always works. However this would fix the js dependencies of BIMvie.ws, but the next thing it will fail on are the API calls (where to send them to?) Maybe this could all be fixed by giving better error messages, for example: 

> BIMvie.ws could not connect to the BIMserver at http://localhost:8080, is your SiteAddress configured correctly?

**Comment by rubendel** [comment]
This has been fixed, the setting is still there, but BIMvie.ws, BIMsurfer 3 and other web modules are not using it anymore. For example for sending emails, it really is necessary, because it's impossible to know on which address(es) a BIMserver is being served (proxy servers etc...).
